### ISSUE TITLE
Lint or report components which are indirectly exported

### ISSUE THREAD

**Issue by brettz9** [vuejs/eslint-plugin-vue#1327]
**Tell us about your environment**

- **ESLint version:** 7.10.0
- **eslint-plugin-vue version:** 7.0.1
- **Node version:** 12.16.2

**The problem you want to solve.**

Currently, exports stated like:

```vue
<script>
const Component = {};

export default Component;
</script>
```

...are not picked up by Vue linting.

**Your take on the correct solution to problem.**

Either the reference could be detected and/or a new rule could be added to *require* anonymous default exports.

**Additional context**

N/A

**Comment by ota-meshi** [comment]
Thank you for this issue.

You can use the special comment `// @vue/component` to mark an object as a Vue component.
https://eslint.vuejs.org/user-guide/#how-does-eslint-detect-components

Also, if you want to force `export default {}`, use `vue/require-direct-export` rule.
https://eslint.vuejs.org/rules/require-direct-export.html

**Comment by brettz9** [comment]
Oh, my apologies--not sure how I missed that. Thanks!
